### PR TITLE
Skittles

### PR DESCRIPTION
**Pull request by cadecairos**
PR, mostly for testing travis. we don't *Actually* need skittles.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "api.webmaker.org",
+  "version": "0.0.0",
+  "description": "Services for Webmaker",
+  "main": "server.js",
+  "scripts": {
+    "start": "node server.js",
+    "test": "npm run test:api && npm run lint",
+    "test:api": "lab -c --verbose --colors --assert code --timeout 5000",
```
Comment (jbuck) [inline package.json]: You should also add `-t 100` which sets the minimum code coverage percentage to 100%

**Comment by jbuck** [comment]
Holy crap this is comprehensive

**Comment by cadecairos** [comment]
Heh

File: README.md [hunk 2]
```diff
@@ -101,23 +103,26 @@ More detail is needed here.
 ## Technical Decisions
 
 #### Database
-- what database? (mysql, postgres, ?)
+- Postgre
```
Comment (davidascher) [inline README.md]: It's called PosgreSQL I believe.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://travis-ci.org/mozilla/api.webmaker.org.svg)](https://travis-ci.org/mozilla/api.webmaker.org)
+
 # api.webmaker.org
```
Comment (thisandagain) [inline README.md]: Would be nice to add some docs here for:
- How to install
- How to configure
- How to run
- How to test

Comment (thisandagain) [inline README.md]: Might even want to remove the roadmap / platform stuff from here too as we are tracking those things in issues.

File: env.sample [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export API_HOST=localhost
+export PORT=2015
+export POSTGRE_CONNECTION_STRING=postgresql://localhost:5432/skittles
+
+## log level - given a chosen level, log events only at or above that level
+## debug < info < warn < error < exception < stat
+export LOG_LEVEL=info
+
+## Log to file - LOG_FILE_PATH and LOG_FILENAME must be defined
```
Comment (jbuck) [inline env.sample]: I have a gut feeling we won't want to use this, and we'll only want to log to stdout, but who knows...

Comment (thisandagain) [inline env.sample]: +1

File: lib/request-log.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+var format = require('util').format;
```
Comment (thisandagain) [inline lib/request-log.js]: Seems like we are re-inventing the wheel here:
https://github.com/trentm/node-bunyan (see: request serialization)
https://github.com/winstonjs/winston
etc.

File: services/api/handlers/skittles.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        if ( err ) {
+          return reply(err);
+        }
+
+        if ( !result.rows.length ) {
+          return reply(Boom.notFound('Can\'t find skittle'));
+        }
+
+        skittle = result.rows[0];
+        console.log( skittle );
```
Comment (thisandagain) [inline services/api/handlers/skittles.js]: Some stay `console.log`s here

**Comment by jbuck** [comment]
@cadecairos I think this is looking pretty solid

File: lib/request-log.js (hunk 5 above)
Comment (cadecairos) [inline lib/request-log.js]: @thisandagain we'd still have to write a serializer for hapi request objects, wouldn't we?

Comment (thisandagain) [inline lib/request-log.js]: Ah... interesting. HAPI overrides the standard http `request` object?

Comment (cadecairos) [inline lib/request-log.js]: I'm trying out https://www.npmjs.com/package/hapi-bunyan to see if it suits our needs

Comment (cadecairos) [inline lib/request-log.js]: @thisandagain I've gotten bunyan logging working.

looks like bunyan doesn't have as much useful logging information as I'd hoped for though

**Pull request merged**